.cursor/rules/sitecore.mdc

```diff
@@ -0,0 +1,256 @@
+import { NoDataFallback } from '@/utils/NoDataFallback';
+
+export const HeroDefault: React.FC<HeroProps> = (props) => {
+  const { fields, isPageEditing } = props;
+  
+  if (!fields?.data?.datasource) {
+    return <NoDataFallback componentName="Hero" />;
+  }
+  
+  const { title, description, image } = fields.data.datasource;
+  
+  // Component implementation
+};
+```
+
+## Field Handling Patterns
+
+Sitecore Field Components:
+- Use `@sitecore-content-sdk/nextjs` field components
+- Access field values through `jsonValue` property
+- Handle optional fields with conditional rendering
+- Use proper semantic HTML tags
+
+```typescript
+import { Text, RichText, Image } from '@sitecore-content-sdk/nextjs';
+
+// Field rendering patterns
+{title?.jsonValue && (
+  <Text
+    tag="h1"
+    field={title.jsonValue}
+    className="hero-title text-4xl font-bold"
+  />
+)}
+
+{description?.jsonValue && (
+  <RichText field={description.jsonValue} />
+)}
+
+{image?.jsonValue && (
+  <Image
+    field={image.jsonValue}
+    alt={title?.jsonValue?.value || 'Hero image'}
+    className="w-full h-auto"
+  />
+)}
+```
+
+## Editing Mode Handling
+
+Page Editing Support:
+- Use `useSitecore` hook to access page mode
+- Check `page.mode.isEditing` for editing state
+- Pass `isPageEditing` prop to variant components
+- Provide different rendering for editing vs. preview
+
+```typescript
+import { useSitecore } from '@sitecore-content-sdk/nextjs';
+
+export const Default: React.FC<ComponentProps> = (props) => {
+  const { page } = useSitecore();
+  const { isEditing } = page.mode;
+  const isEditMode = props.isPageEditing || isEditing;
+  
+  if (isEditMode) {
+    // Simplified rendering for editing mode
+  }
+  
+  // Normal rendering
+};
+```
+
+## Styling and UI Patterns
+
+Tailwind and Shadcn/ui:
+- Use Tailwind CSS utility classes throughout
+- Import UI components from `@/components/ui/`
+- Use `cn()` utility for conditional classes
+- Follow container query patterns with `@container` classes
+
+```typescript
+import { cn } from '@/lib/utils';
+import { Button } from '@/components/ui/button';
+
+<div className={cn(
+  '@container bg-primary rounded-default',
+  'relative mx-auto my-6 max-w-7xl px-4 py-16',
+  isActive && '@md:w-full'
+)}>
+```
+
+## Utility Integration
+
+Common Utilities:
+- Use `NoDataFallback` for missing datasources
+- Import image wrapper: `Default as ImageWrapper` from `@/components/image/ImageWrapper.dev`
+- Use button components: `ButtonBase` from `@/components/button-component/ButtonComponent`
+- Implement localization with `useI18n` and `dictionaryKeys`
+
+```typescript
+import { NoDataFallback } from '@/utils/NoDataFallback';
+import { Default as ImageWrapper } from '@/components/image/ImageWrapper.dev';
+import { ButtonBase } from '@/components/button-component/ButtonComponent';
+import { useI18n } from 'next-localization';
+import { dictionaryKeys } from '@/variables/dictionary';
+```
+
+## Performance Patterns
```

Comment:
Can you please add some description for `@sitecore-content-sdk/nextjs` imports and when they could be used?
I.e. imports from `nextjs/config-cli` submodule should not be used in client context (i.e. next config file, components) as they use server-only logic etc